### ISSUE TITLE
Web - Invitation bottom link leads to magic code expired page when invited in a room

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#52546]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.62-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Issue was found when executing this PR:** https://github.com/Expensify/App/pull/46052
**Email or phone of affected tester (no customers):** nathanmulugetatesting+2083@gmail.com
**Issue reported by:** Applause Internal Team

## Action Performed:
1. Navigate to staging.new.expensify.com and login as user A
2. Create a room (user A)
3. Invite a non existent email address into the room (user A invites a user B)
4. As the invitee wait for the email (user B)
5. Open the email and click on the bottom link (user B)

## Expected Result:
App signs user in and takes user to the room

## Actual Result:
Magic code expired page shown

## Workaround:
Unknown

## Platforms:

- [ ] Android: Standalone
- [ ] Android: HybridApp
- [ ] Android: mWeb Chrome
- [ ] iOS: Standalone
- [ ] iOS: HybridApp
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/47dd0bf2-5c9c-4723-aef2-b6b6963e00d1
 
</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Issue Owner</summary>Current Issue Owner: @deetergp</details>

**Comment by RachCHopkins** [comment]
@IuliiaHerets I don't understand the repro steps:

> 3. Invite a non existent email address into the room
> 4. As the invitee wait for the email

If the email does not exist, who is the invitee who is waiting?

**Comment by IuliiaHerets** [comment]
@RachCHopkins I updated the steps a bit for better understanding. 
1. Navigate to staging.new.expensify.com and login as user A
2. Create a room (user A)
3. Invite a non existent email address into the room (user A invites a user B)
4. As the invitee wait for the email (user B)
5. Open the email and click on the bottom link (user B)

**Comment by mountiny** [comment]
cc @blazejkustra @deetergp https://github.com/Expensify/App/pull/46052

**Comment by blazejkustra** [comment]
Thanks @mountiny for bumping 😄 

I think it is likely a backend issue, and it was known before merging ([here](https://github.com/Expensify/App/pull/46052#issuecomment-2477063168) more context). @deetergp are you going to investigate this?

**Comment by RachCHopkins** [comment]
@IuliiaHerets by non-existent email, do you mean "a legitimate email address that does not already have an Expensify account associated with it" as opposed to "an email address that is fictional and does not exist"?

**Comment by IuliiaHerets** [comment]
@RachCHopkins Yes, a user with a legitimate email address that does not already have an Expensify account

**Comment by RachCHopkins** [comment]
Just assigning you here @deetergp so that you don't miss this!

**Comment by deetergp** [comment]
I'm taking an educated guess that this is part of #convert since it's related to invites and guided setups.

**Comment by deetergp** [comment]
After some digging into this today, I've discovered that the reason we're getting this "magic code expired" message is because we are not creating a `validateCode` for these users when they are created. @marcaaron I wonder if that has something to do with the recent changes to `addComment` that we were just talking about. That is where I'm going to check next…

**Comment by deetergp** [comment]
I stand corrected. We don't create the validateCode till after a user has logged in. But when comparing a user created by a room invite, which doesn't work right now, to a DM invite, which sorta works right now, the DM invited user's account is `validated: 0`, but the room invited user is `validated: 1`. I suspect that we are erroneously marking the room invited users as validated and that's what is causing the problem. Will dig more into it on Monday.

**Comment by RachCHopkins** [comment]
Would we call this `Internal` @deetergp?

**Comment by deetergp** [comment]
Oh yes, I'm pretty certain this one's internal.

**Comment by deetergp** [comment]
Circling back to this, I've confirmed that it's validation that's the issue. These users should be closed and unvalidated until they respond to the invitation links. The steps I used to prove it were:
1. As User A, invite a new user with a valid email address but no Expensify account, to a room.
2. Verify that the user's account was created closed, but also `validated: 1` in the data base.
3. Changed their validation status to `validated: 0` with a CQ
4. Found the invitation in the user's inbox.
5. Clicked on the link
6. Verified that the new user was taken to the room and not a 404 page

Now all we need to determine why these users are created in a validated state…
